fastify.js

```diff
@@ -70,7 +83,31 @@ function build (options) {
       return
     }
 
-    router(stripUrl(req.url), req, res)
+    runInSeries(
+      new State(req, res),
+      _onRequestIterator,
+      onRequestRaw(),
+      routeCallback
+    )
+  }
+
+  function State (req, res) {
+    this.req = req
+    this.res = res
+  }
+
+  function _onRequestIterator (fn, cb) {
+    setImmediate(fn, this.req, this.res, cb)
```

Comment:
Yes, but this adds a `setImmediate` for every hook executed. I think we should avoid this, and just retain the only `setImmediate` we have later on.